rules/do-you-use-these-useful-react-hooks/rule.md

```diff
@@ -1,161 +1,265 @@
-
-5. `useReducer` Hook
-
-    `useReducer` is a React Hook that allows you to manage state in a more complex way than useState. It is based on the reducer pattern from functional programming and takes in a reducer function and an initial state value as arguments. The reducer function takes in the current state and an action and returns a new state.
-    
-    ```javascript
-    import React, { useReducer } from 'react';
-    
-    function counterReducer(state, action) {
-      switch (action.type) {
-        case 'increment':
-          return { count: state.count + 1 };
-        case 'decrement':
-          return { count: state.count - 1 };
-        default:
-          throw new Error();
-      }
-    }
-    
-    function Counter() {
-      const [state, dispatch] = useReducer(counterReducer, { count: 0 });
-    
-      return (
-        <div>
-          <p>Count: {state.count}</p>
-          <button onClick={() => dispatch({ type: 'increment' })}>Increment</button>
-          <button onClick={() => dispatch({ type: 'decrement' })}>Decrement</button>
-        </div>
-      );
-    }
-    ```
-    
-    In the example above, useReducer is used to create a state variable called state and a dispatch function called dispatch. The counterReducer function is used as the reducer function. The initial state is an object with a count property set to 0. The Counter component uses the state.count value to render the current count, and the dispatch function to update the count by dispatching actions with the type property set to either 'increment' or 'decrement'. The reducer function updates the state based on the action dispatched by returning a new state object with the updated count value.
+## 1️⃣ **useState**: Managing Local State 🧠
+
+The `useState` hook lets you add state to functional components. Call `useState` at the top level of your component to declare one or more state variables.
+
+```jsx
+import { useState } from 'react';
+
+export default function Counter() {
+  const [count, setCount] = useState(0);
+
+  function handleClick() {
+    setCount(count + 1);
+  }
+
+  return (
+    <button onClick={handleClick}>
+      You pressed me {count} times
+    </button>
+  );
+}
+```
+**Figure: Using useState for a counter component**
+
+::: info
+***Naming Convention***:
+It's a common convention to name state variables using the pattern [**count**, **setCount**] with array destructuring.
+:::
+
+`useState` returns an array with exactly two items:
+
+1. The current state of this state variable, initially set to the initial state you provided.
+2. A function that lets you update its value.
+
+### ✅Recommended Usage
```

Comment:
1. Please add a space between emoji and text

https://www.ssw.com.au/rules/use-emojis/#:~:text=Tip%3A%20Always%20remember%20to%20add,and%20text%2C%20for%20better%20readability.